tools/rpi-sign-bootcode

```diff
@@ -196,34 +230,50 @@ def main():
         2711 mode:
         rpi-sign-bootcode --debug -c 2711 -i bootcode.bin.clr -o bootcode.bin -k 2711_rsa_priv_0.pem -n 0 -m bootcode-production.key
 
-        2712 C1 and D0 mode:
-          * HMAC not included on 2712
+        2712 A0, B0 mode:
+        rpi-sign-bootcode --debug -c 2712 -i bootcode.bin.clr -o bootcode.bin -k 2712_rsa_priv_0.pem -n 0 -m bootcode-production.key
+
+        2712 B1 mode:
+          * HMAC not included
           * RSA public key included - ROM just contains the hashes of the RPi public keys.
 
-        Customer counter-signed signed:
+        rpi-sign-bootcode --debug -c 2712 -s B1 -i bootcode.bin.clr -o bootcode.bin -k 2712_b1_rsa_priv_0.pem -n 0 -v 0
+
+        Customer signed:
           * Exactly the same as Raspberry Pi signing but the input is the Raspberry Pi signed bootcode.bin
           * The key number will probably always be 16 to indicate a customer signing
 
         rpi-sign-bootcode --debug -c 2712 -i bootcode.bin.sign2 -o bootcode.bin -k customer.pem
+
+        PKCS#11 - HSM wrapper:
+          * hsm-wrapper takes a single argument which is a temporary filename containing the data to sign.
+          * hsm-wrapper outputs the PKCS#1 v1.5 signature in base64 format 
+          * There is no facility to pass the private key, custom HSM arguments - the caller should generate a dedicated wrapper script.
+          * The public key in PEM format MUST be specified.
+
+        rpi-sign-bootcode --debug -c 2712 -i bootcode.bin.sign2 -o bootcode.bin -p public.pem -H hsm-wrapper
     """
     parser = argparse.ArgumentParser(help_text)
     parser.add_argument('-o', '--output', required=False, help='Output filename . If not specified the signed images is written to stdout in base64 format')
     parser.add_argument('-c', '--chip', required=True, type=int, help='Chip number')
```

Comment:
Not directly part of this PR, but probably doesn't hurt to expand the help text to `help='Chip number (2711 or 2712)'`

Reply:
NGTF